collections/_article/rendering-priorities-january-2024.md

```diff
@@ -0,0 +1,116 @@
+categories: ["progress-report"]
+author: Clay John
+image: /storage/blog/covers/rendering-priorities-jan-2024.webp
+date: 2024-01-11 20:00:00
+---
+
+We released [Godot 4.2](https://godotengine.org/article/godot-4-2-arrives-in-style/) on November 30 2023  with many exciting features and countless fixes and improvements. Godot 4.2 is starting to look like what we had envisioned for Godot 4. Accordingly, we are spending more time thinking and talking about what the future holds for the [rendering team](https://godotengine.org/teams/#rendering).
+
+At GodotCon this year I spoke about new features that the rendering team would like to see implemented in the future. While not all of these features will make it in, it's a pretty good overview of what we are thinking and talking about.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/MW3IFMvDTCY" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+Also, a reminder for those interested in contributing, the rendering team meets weekly and all are welcome to join the meetings ([calender link](https://calendar.google.com/calendar/u/0/r?cid=dXBwOGIwZXU0a3BlZjFjNTB2dTJmM2tjOGNAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ)).
+
+Also come join us in the `#rendering` channel on [chat.godotengine.org](https://chat.godotengine.org)!
+
+## Update on previous priorities and 4.2
+
+The below are the priorities we identified in [July 2023](https://godotengine.org/article/rendering-priorities-july-2023) with a short description of their current state.
+
+### Performance
+
+**1. Split Dynamic/Static Light3D shadows (on hold)**
+
+This work has been put on hold for the time being. We haven't been able to implement it in a way that sufficiently improves performance enough to justify the additional complexity to user's workflow and to our codebase. We intend to return to this in the future.
+
+**2. Background pipeline compilation (in progress)**
+
+This remains a high priority and has only been pushed back due to other foundational work on the RenderingDevice backend.
+
+**3. Shader compilation groups ([GH-79606](https://github.com/godotengine/godot/pull/79606)) (done)**
+
+We completed this and it was included in the 4.2 release! As a reminder, this optimization decreases shader compilation time for most 3D materials and should result in faster startup times and less stuttering in many games.
+
+### Stability
+
+**1. BUGS (in progress, always)**
+
+We fixed over [200](https://github.com/godotengine/godot/pulls?q=is%3Apr+milestone%3A4.2+is%3Aclosed+label%3Atopic%3Arendering%2Ctopic%3Ashaders%2Ctopic%3Aparticles+label%3Abug) rendering related bugs in 4.2! 
+
+**2. Proper multi-threading in the RenderingDevice (in progress)**
+
+This was delayed due to foundational work on the RenderingDevice backend. But the work is currently underway and is showing promising [results](https://x.com/dariosamo/status/1735363059202375819?s=20) already!
```

Comment:
First `x.com` link, is this the end of the resistance? :P

Reply:
:( The tweet is too good to not link lol

Reply:
I just meant I would usually link `twitter.com` ;)

Reply:
Oh, I didn't realize that still works
```suggestion
This was delayed due to foundational work on the RenderingDevice backend. But the work is currently underway and is showing promising [results](https://twitter.com/dariosamo/status/1735363059202375819) already!
```